### PR TITLE
Flake example in README

### PR DESCRIPTION
**Pull request by MMesch**
I added a simple flake example directly to the README. We should probably also add how this can be initialized directly with `nix flake init -t`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +80,58 @@ This process can be largely accelerated by using [cachix](https://cachix.org):
...
+      };
+
+      jupyterEnvironment = pkgs.jupyterlabWith {
+          kernels = [ iPython iHaskell ];
+      };
+    in
+    {
+#      packages.x86_64-linux.jupyterLab = jupyterEnvironment;
+#      defaultPackage.x86_64-linux = jupyterEnvironment;
+      apps.x86_64-linux.jupterlab = {
```
Comment (guaraqe) [inline README.md:126]: This is correct?

File: README.md [hunk 2]
```diff
@@ -78,6 +80,58 @@ This process can be largely accelerated by using [cachix](https://cachix.org):
...
+        ignoreCollisions = true;
+      };
+
+      jupyterEnvironment = pkgs.jupyterlabWith {
+          kernels = [ iPython iHaskell ];
+      };
+    in
+    {
+#      packages.x86_64-linux.jupyterLab = jupyterEnvironment;
+#      defaultPackage.x86_64-linux = jupyterEnvironment;
```
Comment (guaraqe) [inline README.md:125]: What are these supposed to do?

File: README.md (hunk 1 above)
Comment (guaraqe) [reply README.md:126]: Same for two lines below.

**Review by guaraqe — APPROVED** [review]

**Comment by guaraqe** [comment]
I have some questions, but in principle having this example is good.

**Comment by SenchoPens** [comment]
Hi, I fixed the example a bit:

```nix
{
  description = "JupyterLab Flake";

  inputs = {
      jupyterWith.url = "github:tweag/jupyterWith";
      flake-utils.url = "github:numtide/flake-utils";
  };

  outputs = { self, nixpkgs, jupyterWith, flake-utils }:
    flake-utils.lib.eachDefaultSystem (system:
      let
        pkgs = import nixpkgs {
          system = system;
          overlays = nixpkgs.lib.attrValues jupyterWith.overlays;
        };

        iHaskell = pkgs.kernels.iHaskellWith {
          name = "haskell";
          packages = p: with p; [ hvega ];
        };

        iPython = pkgs.kernels.iPythonWith {
          name = "Python-env";
          packages = p: with p; [ sympy numpy ];
          ignoreCollisions = true;
        };

        jupyterEnvironment = pkgs.jupyterlabWith {
          kernels = [ iPython iHaskell ];
        };
      in rec {
        apps.jupterlab = {
          type = "app";
          program = "${jupyterEnvironment}/bin/jupyter-lab";
        };
        defaultApp = apps.jupterlab;
      }
    );
}
```

P.S. I think that it will be best to include the iHaskell kernel commented as it's heavy.

File: README.md (hunk 1 above)
Comment (MMesch) [reply README.md:126]: yes, but we should probably uncomment all three lines. Then you can do `nix run` and `nix build`.

File: README.md (hunk 2 above)
Comment (MMesch) [reply README.md:125]: They are for `nix build`, but yea they should probably be uncommented (or removed, couldn't decide).

Comment (SenchoPens) [reply README.md:125]: I think the best way is to remove this in the README flake to keep it short, but keep it in [example/flake/flake.nix](https://github.com/tweag/jupyterWith/blob/master/example/flake/flake.nix), and mention this file in the README.md as the full flake.

**Comment by guaraqe** [comment]
I used the example suggested by @SenchoPens

**Pull request merged**